hack/jenkins/e2e.sh

```diff
@@ -93,6 +97,22 @@ GCE_DEFAULT_SKIP_TESTS=(
     "Example"
     )
 
+# Tests which cannot be run on GKE, e.g. because they require
+# master ssh access.
+GKE_REQUIRED_SKIP_TESTS=(
+    "Nodes"
+    "Etcd\sFailure"
+    "MasterCerts"
+    "Shell"
+    )
+
+GKE_REBOOT_SKIP_TESTS=(
```

Comment:
Right, we want to run a small, unflaky set of tests along with reboot to exercise cluster functionality post-reboot. Refactored a bit to remove duplication.